client/my-sites/hosting/github/github-connect-card/index.tsx

```diff
@@ -1,16 +1,106 @@
-import { Card } from '@automattic/components';
-import { useI18n } from '@wordpress/react-i18n';
+import { Button, Card, Spinner } from '@automattic/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
 import CardHeading from 'calypso/components/card-heading';
-import iconGitHub from '../github.svg';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormSelect from 'calypso/components/forms/form-select';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import SocialLogo from 'calypso/components/social-logo';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { RepoBranch, useGithubBranches } from './use-github-branches';
+import { Repo, useGithubRepos } from './use-github-repos';
+
+import './style.scss';
 
-// todo - just a placeholder for now as the implementation of this
-// component falls under another PR.
 export const GithubConnectCard = () => {
-	const { __ } = useI18n();
+	const siteId = useSelector( getSelectedSiteId );
+
+	const [ selectedRepo, setSelectedRepo ] = useState< Repo >();
+	const [ selectedBranch, setSelectedBranch ] = useState< RepoBranch >();
+	const { data: repos, isLoading: isLoadingRepos } = useGithubRepos( siteId );
+	const { data: branches, isLoading: isLoadingBranches } = useGithubBranches(
+		siteId,
+		selectedRepo?.name
+	);
+
+	useEffect( () => {
+		if ( repos ) {
+			setSelectedRepo( selectedRepo || repos[ 0 ] );
+		}
+		if ( branches ) {
+			setSelectedBranch( branches[ 0 ] );
+		}
+	}, [ repos, branches, selectedRepo ] );
+
+	const showSpinner = ! repos && ! branches && ( isLoadingRepos || isLoadingBranches );
+
+	const handleRepoSelect = ( repoName: string ) => {
+		const selectedRepo = repos?.find( ( repo ) => repo.full_name === repoName );
+		setSelectedRepo( selectedRepo );
+	};
+
+	const handleBranchSelect = ( branchName: string ) => {
+		const selectedBranch = branches?.find( ( branch ) => branch.name === branchName );
+		setSelectedBranch( selectedBranch );
+	};
+
 	return (
-		<Card className="github-hosting-card">
-			<img className="github-hosting-icon" src={ iconGitHub } alt="" />
+		<Card className="connect-branch-card">
+			<SocialLogo className="material-icon" icon="github" size={ 32 } />
 			<CardHeading>{ __( 'Connect Branch' ) }</CardHeading>
+			<div>
+				<p>{ __( 'Changes pushed to connected branch will be deployed automatically' ) }</p>
+				{ ! showSpinner ? (
```

Comment:
Personal preference, but I think it's more readable to put the shorter ternary branch first.

```suggestion
				{ showSpinner ? <Spinner /> : (
```